File: app/src/main/java/rubik/shifttest/presentation/GreetingDialog.java
package rubik.shifttest.presentation;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatDialogFragment;

public class GreetingDialog extends AppCompatDialogFragment {

    private String mGreeting;
    private static final String GREET = "Hello";

    GreetingDialog(String greeting) {
        mGreeting = greeting;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        return builder.setTitle(GREET).setMessage(mGreeting).create();
    }

}

File: feature/user/greeting/src/main/kotlin/ru/rubik/greeting/ui/GreetingDialog.kt
package ru.rubik.greeting.ui

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AppCompatDialogFragment

class GreetingDialog(private val nameToGreet: String) : AppCompatDialogFragment() {

	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

		return AlertDialog.Builder(activity).setTitle("Hello").setMessage(nameToGreet).create()
	}
}